kong/core/balancer.lua

```diff
@@ -309,6 +310,7 @@ local function execute(target)
 
   target.ip = ip
   target.port = port
+  target.hostname = target.host
```

Comment:
`host` is the hostname extracted from the upstream url (and might be an upstream-name)
`hostname` is the name that matches the final ip (never an upstream-name)
The original `host` field must be retained in case of a retry, whereas `hostname` might get a different value for each try.